File: 20220810/모범답안/20220810_09.js/LeastRepresenter.py
from functools import reduce


def solution(arr):
    """
    :param arr: int[]
    :return: int
    """

    arr.sort()

    representers = []
    for i in range(int(len(arr) / 2)):
        curMinIndex = i * 2
        representers.append(max(arr[curMinIndex], arr[curMinIndex + 1]))
    return reduce(lambda pre, cur: pre + cur, representers, 0)

File: 20220810/모범답안/20220810_09.js/LeastRepresenter.js
/**
 * @param arr {number[]}
 * @returns {number}
 */
function solution(arr) {
  arr.sort((a, b) => a - b)

  const representers = []
  for (let i = 0; i < arr.length / 2; i++) {
    const curMinIndex = i * 2
    representers.push(Math.max(arr[curMinIndex], arr[curMinIndex + 1]))
  }
  return representers.reduce((pre, cur) => pre + cur, 0)
}

solution